tests/acceptance/ZzzReinstallFromScratchCest.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace MailPoet\Test\Acceptance;
+
+use MailPoet\Models\Setting;
+use MailPoet\Test\DataFactories\Form;
+use MailPoet\Test\DataFactories\Newsletter;
+use MailPoet\Test\DataFactories\Segment;
+
+require_once __DIR__ . '/../DataFactories/Form.php';
+require_once __DIR__ . '/../DataFactories/Newsletter.php';
+require_once __DIR__ . '/../DataFactories/Segment.php';
+
+// This test should run last, because it removes all test data
+class ZzzReinstallFromScratchCest {
+
+  function reinstallFromScratch(\AcceptanceTester $I) {
+    $I->wantTo('Reinstall from scratch');
+    $I->login();
+
+    // Step 1 - create email, form, list, but no new subscribers, since they are imported in previous tests
+    $newsletter = new Newsletter();
+    $newsletter->create();
+    $form = new Form();
+    $form->create();
+    $segment = new Segment();
+    $segment->create();
+
+    // Step 2 - reinstall from scratch
+    $I->amOnPage('/wp-admin/admin.php?page=mailpoet-settings#advanced');
+    $I->waitForElement('#mailpoet_reinstall', 5);
+    $I->click('Reinstall now...');
+    $I->acceptPopup();
+    $I->waitForElement('#mailpoet_loading', 5);
+    $I->waitForElementNotVisible('#mailpoet_loading', 10);
+
+    // Step 3 - skip intro
+    Setting::setValue('show_intro', 0);
+
+    // Step 4 - check if data are emptied and repopulated
+    // Check emails
+    $I->amOnMailpoetPage('Emails');
+    $I->waitForText('Nothing here yet!', 5);
+    $I->seeNumberOfElements('[data-automation-id^=listing_item_]', 0);
+    // Check forms
+    $I->amOnMailpoetPage('Forms');
+    $I->waitForText('A GDPR friendly form', 5, '[data-automation-id="listing_item_1"]');
+    $I->seeNumberOfElements('[data-automation-id^=listing_item_]', 1);
+    // Check lists
+    $I->amOnMailpoetPage('Lists');
+    $I->waitForText('WordPress Users', 5, '[data-automation-id="listing_item_1"]');
+    $I->see('My First List', '[data-automation-id="listing_item_2"]');
+    $I->seeNumberOfElements('[data-automation-id^=listing_item_]', 2);
+    // Check subscribers
+    $I->amOnMailPoetPage('Subscribers');
+    $I->waitForText('admin', 5, '[data-automation-id="listing_item_1"]');
+    $I->seeNumberOfElements('[data-automation-id^=listing_item_]', 20); // admin + 40 test users, but default paging is set to 20
```

Comment:
This test depends on seeing some number of subscribers in the subscribers list, but does not create any by itself to ensure subscribers get removed correctly as well.

The second point is that this test expects 20 WordPress user subscribers in the test, which I presume is due to WP users being created in other tests.
Because if we remove the `Zzz` test-order-changing part of the class name, here we see only `2` subscribers/WP users.
This test is very brittle, because what happens if other tests get added that create more WP users? This number will then change as well, which means another implicit relationship.

After reinstallation, only WP user subscribers should remain, so perhaps there is a better way to check that.
E.g. that the number of subscribers actually matches the number of WP users, or that in the list filter only "WordPress Users" list has more than one subscriber, or that a subscriber you've created earlier in the test got removed.